### ISSUE TITLE
Switching back and forth between accounts makes the app crash

### ISSUE THREAD

**Issue by TheSonAlsoRises** [QuantumBadger/RedReader#241]
I have two accounts set up. If I switch to the other, then immediately back to the original, RedReader force stops.

**Comment by hexid** [comment]
Just to give a few more details on this, it appears as though it only occurs on the main menu.

It also seems to only be an issue if the current account's subreddit list is empty (searching for it or connection error).

I was only able to reproduce this once through switching between accounts normally. After that, I could cause the crash by disabling my network, refreshing the subreddit list, and then trying to switch accounts.

```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.TypedArray android.content.Context.obtainStyledAttributes(int[])' on a null object reference
at org.quantumbadger.redreader.adapters.AccountListAdapter.<init>(AccountListAdapter.java:43)
at org.quantumbadger.redreader.fragments.AccountListDialog$5.run(AccountListDialog.java:217)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5294)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
```

**Comment by QuantumBadger** [comment]
Thanks @TheSonAlsoRises for the report and @hexid for the extra info. I've also noticed this happening myself. At first glance it looks like the result of `getActivity()` we're getting in `onRedditAccountChanged()` is null -- it should be possible to fix this by storing the activity in `onCreateDialog()` and using that instead of `getActivity()`.

**Comment by QuantumBadger** [comment]
Should be fixed in v1.9.1.
